lib/remote/apiuser.cpp

```diff
@@ -34,3 +50,47 @@ ApiUser::Ptr ApiUser::GetByClientCN(const String& cn)
 
 	return nullptr;
 }
+
+bool ApiUser::ComparePassword(String password) const {
+	String other_passwd = CreateHashedPasswordString(password, this->m_Salt, false);
+
+	const char *p1 = other_passwd.CStr();
+	const char *p2 = this->m_Hashed_passwd.CStr();
+
+	volatile char c = 0;
+
+	for (size_t i=0; i<64; ++i)
+		c |= p1[i] ^ p2[i];
+
+	return (c == 0);
+}
```

Comment:
Is this worthwhile to be moved to libbase into the Utility or TlsUtility class and as such passing all required params?